### DISCUSSION TITLE
How do I apply a margin to all (except the first) children elements with Tailwind CSS?

### DISCUSSION THREAD

**Discussion by wimdeblauwe** [tailwindlabs/tailwindcss#2156] [Help] [answered]
How do people handle applying a CSS class to all elements except the first one?

Example:

```
<ul class="list-disc pl-5">
          <li>
            Your password must be at least 8 characters
          </li>
          <li class="mt-1">
            Your password must included at least one pro wrestling finishing move
          </li>
</ul>
```

Each `<li>` should have `mt-1`, but not the first one.

If you try to use this with a templating language (I use Thymeleaf), I need to do something like:

```
<ul class="list-disc pl-5">
    <li th:each="err,iter : ${errors}"
         th:classappend="${iter.index > 0 ?'mt-1':''}">
</ul>
```

Are there other (better) ways to handle this?

I wonder if for example something like this would be possible:

```
<ul class="list-disc pl-5">
    <li th:each="err: ${errors}"
         class="not-first:mt-1">
</ul>
```

**Comment by adamwathan** [comment, marked as answer]
For this specific problem I'd recommend our [space between utilities](https://tailwindcss.com/docs/space/#app) 👍 

We also have [divide utilities](https://tailwindcss.com/docs/divide-width/#app) for doing the same thing with borders.

I find these two utilities cover 99% of cases I'd use `not-first` for, and in the other cases I'd use conditional templating logic like you mentioned.

**Reply by wimdeblauwe** [reply]
Thanks! I took the example from Tailwind UI. Does it mean those examples should be updated to use those instead?

**Reply by simonswiss** [reply]
We're working on updating the Tailwind UI components to use these spacing utilities indeed! 👍

**Reply by AbhishekSChauhan** [reply]
Yes `space between utilities` can be used for adding margin to all-but-the-first-item in a group but it has limitations for grids and flex which can not be solved using `gap` when elements are being generated in some kind of loop, and in JIT engine support for styling pseudo-elements like `before, after, first-letter,first-line, marker, and selection` but `not-first`.
Is there any other utility we can utilise for `not-first`.

**Reply by asmimo** [reply]
I also did not find `not-first` so here's the plugin i wrote

```js
const plugin = require('tailwindcss/plugin')

module.exports = {
  plugins: [
    plugin(function ({ addVariant, e }) {
      addVariant('not-first', ({ modifySelectors, separator }) => {
        modifySelectors(({ className }) => {
          return `.${e(`not-first${separator}${className}`)}:not(:first-child)`
        })
      })
    }),
  ]
}
```

**Reply by acidjazz** [reply]
I would choose this plug-in answer over the recommended space between and divide utilities, you cant use them together properly.  I think not-first should be in tailwind

**Reply by simonswiss** [reply]
Honestly, since the original answer, the `gap` property in Flex has become well-supported, and for most use cases, you can use that instead (Flex wrapper with gap spacer).

**Reply by acidjazz** [reply]
@simonswiss the problem is even w/ a gap and a divide the line is either on the top or the bottom, you cant have your dividing line in the middle which is what most cases want, you have to add a pt-8 or py-8 and that 1st row is gonna need to not have it.

**Reply by simonswiss** [reply]
Ah yes, sorry I totally skipped the "divide" part of the equation 😅
